trpcultivate_phenoshare/src/Plugin/TripalImporter/TripalCultivatePhenoshareImporter.php

```diff
@@ -474,60 +630,87 @@ public function formValidate($form, &$form_state) {
+          // Metadata Validation
+          // *******************************************************************
+          foreach ($validators['metadata'] as $validator_name => $validator) {
+            // Set failures for this validator name to an empty array to signal
+            // that this validator has been run.
+            $failures[$validator_name] = [];
+            // Validate metadata input value.
+            $result = $validator->validateMetadata($form_values);
+
+            // Check if validation failed and save the results if it did.
+            if (array_key_exists('valid', $result) && $result['valid'] === FALSE) {
+              $failed_validator = TRUE;
+              $failures[$validator_name] = $result;
+            }
+          }
 
-          foreach ($scopes as $scope) {
-            // Create instance of the scope-specific plugin and
-            // perform validation.
-            $validator = $manager->getValidatorIdWithScope($scope);
-            $instance = $manager->createInstance($validator);
+          $validation_feedback = $this->processValidationMessages($failures);
 
-            // Set other validation level to upcoming/todo if a
-            // validation failed.
-            $skip = ($failed_validator > 0) ? 1 : 0;
+          // Save all validation results in Drupal storage to create a
+          // summary report.
+          $storage = $form_state->getStorage();
+          $storage[$this->validation_result] = $validation_feedback;
+          $form_state->setStorage($storage);
 
-            // Load values.
-            $instance->loadAssets($project, $genus, $file, $headers, $skip);
+          // Check if the $validation_feedback contains 'fail' or 'todo' status.
+          // If either is found, prevent form submission.
+          $submit_form = TRUE;
 
-            // Perform current scope level validation.
-            $validation[$scope] = $instance->validate();
+          foreach ($validation_feedback as $feedback_item) {
+            if ($feedback_item['status'] == 'todo' || $feedback_item['status'] == 'fail') {
+              $submit_form = FALSE;
 
-            // Inspect for any failed validation to halt the importer.
-            if ($validation[$scope]['status'] == 'fail') {
-              $failed_validator++;
+              // No need to inspect other validators, a single instance of
+              // fail/todo is sufficient to prevent form submission.
+              break;
             }
           }
 
-          // Save all validation results in Drupal storage to be used by
-          // validation window to create summary report.
-          $storage = $form_state->getStorage();
-          $storage[$this->validation_result] = $validation;
-          $form_state->setStorage($storage);
+          if ($submit_form === FALSE) {
+            // Provide a general error message indicating that input values and/or the
+            // data file may contain one or more errors.
```

Comment:
Line 707 exceeded 80 chars
```suggestion
            // Provide a general error message indicating that input values 
            // and/or the data file may contain one or more errors.
```

Reply:
The code suggestion caused a 'space at the end of line' Problem and is fixed in the most recent commit.